src/components/form-elements/Select.jsx

```diff
@@ -115,11 +128,22 @@ const Select = (props: SelectPropsType) => {
     },
     className
   );
-  const optionsElements = options.map(({value, text}) => (
-    <option key={value} value={value}>
-      {text}
-    </option>
-  ));
+
+  const optionsElements = options.map(item => {
+    if (typeof item.label === 'string' && Array.isArray(item.options)) {
+      return (
+        <optgroup key={item.label} label={item.label}>
```

Comment:
@coderitual I think I need your help here :)
After checks in line 133, flow still do not distinguish which array element type is here - should be GroupedOptionsPropsType 

<img width="1080" alt="Screenshot 2021-06-25 at 09 47 40" src="https://user-images.githubusercontent.com/13717523/123382278-6a4c5780-d59a-11eb-9d0a-407bce23e037.png">

[Here](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgDyOGAlnAHYAquBAvGAN6phgZ4AeGAXGAM4YATqQoBzANwswANwCGMAK55eA4WMkBfdNnxgA4oLgL8AE2JlKNXQ2asYsgEZ4YKoSIlS4JchT68AgoKCslgAPOY+VngAfJratETelHxRYAyBwWERlgkAPvqGxnhmSdS00egAxskYYJyyALY4MHjZvmlgANpSjHKKymAA5F4WFACMAEwAzIMANGycPENtY4Masz19SrzDpRMTcwtcOwDKzniVtXznl3CCaxustqxg9k4uQwAyjs5HGHNSVgjHx+LqAl69eTbIbAyiTKYAFmmh3Yx2WpVW63BTy2A12ZGmCMo+xRi1ONyuFLuD3BAF1HmAsTioXjYRRpjN5qiloMzi1LvwqfcsbTJKhqr5ag1ZDhTG0+B0ABRs0HyqIASjS0SYUkEeAwCkEFDAKoAdNKcIrSOwGlqdS8wKQoIqdHg4FBHTbTW9fnQ-UNVO5BmAAGQhsAZEKm0h8SNYK1elXqzXPB16g1Gz14Bren4wMDgrTgp1gBPZ03c0Ph63l3Ep7FgdOG42ph1-Xg1nPchltmQsjte3E9l5aB2jl5NzMUBQwGDoVgadWoDRAA) is an example in flows playground,  but there is everything ok.
I was struggling with this some time and for now dont have any idea how to fix it.

Reply:
@ensisf Thanks for example in playground, it might be related to outdated version of flow. I will take a look at your branch and get back to you.

Reply:
Hey, there is some problem with understanding `typeof` in the flow version we're using. This probably was introduced later. also checking isArray is redundant in the code and example in playground.

Reply:
Also we can't use `label` in key here. Label is not unique, maybe combining label plus index will be better here.

Reply:
@coderitual Thanks for helping! All changes are applied.